File: SDKs/googleapis.com/beyondcorp/v1alpha/go/pkg/models/operations/beyondcorpprojectslocationsappconnectionsresolve.go
package operations

import (
	"openapi/pkg/models/shared"
)

type BeyondcorpProjectsLocationsAppConnectionsResolvePathParams struct {
	Parent string `pathParam:"style=simple,explode=false,name=parent"`
}

type BeyondcorpProjectsLocationsAppConnectionsResolveQueryParams struct {
	DollarXgafv    *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
	AccessToken    *string           `queryParam:"style=form,explode=true,name=access_token"`
	Alt            *shared.AltEnum   `queryParam:"style=form,explode=true,name=alt"`
	AppConnectorID *string           `queryParam:"style=form,explode=true,name=appConnectorId"`
	Callback       *string           `queryParam:"style=form,explode=true,name=callback"`
	Fields         *string           `queryParam:"style=form,explode=true,name=fields"`
	Key            *string           `queryParam:"style=form,explode=true,name=key"`
	OauthToken     *string           `queryParam:"style=form,explode=true,name=oauth_token"`
	PageSize       *int64            `queryParam:"style=form,explode=true,name=pageSize"`
	PageToken      *string           `queryParam:"style=form,explode=true,name=pageToken"`
	PrettyPrint    *bool             `queryParam:"style=form,explode=true,name=prettyPrint"`
	QuotaUser      *string           `queryParam:"style=form,explode=true,name=quotaUser"`
	UploadType     *string           `queryParam:"style=form,explode=true,name=uploadType"`
	UploadProtocol *string           `queryParam:"style=form,explode=true,name=upload_protocol"`
}

type BeyondcorpProjectsLocationsAppConnectionsResolveSecurity struct {
	Oauth2  shared.SchemeOauth2  `security:"scheme,type=oauth2"`
	Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
}

type BeyondcorpProjectsLocationsAppConnectionsResolveRequest struct {
	PathParams  BeyondcorpProjectsLocationsAppConnectionsResolvePathParams
	QueryParams BeyondcorpProjectsLocationsAppConnectionsResolveQueryParams
	Security    BeyondcorpProjectsLocationsAppConnectionsResolveSecurity
}

type BeyondcorpProjectsLocationsAppConnectionsResolveResponse struct {
	ContentType                                                             string
	GoogleCloudBeyondcorpAppconnectionsV1alphaResolveAppConnectionsResponse *shared.GoogleCloudBeyondcorpAppconnectionsV1alphaResolveAppConnectionsResponse
	StatusCode                                                              int64
}

File: SDKs/googleapis.com/beyondcorp/v1alpha/typescript/src/sdk/models/operations/beyondcorpprojectslocationsappconnectionsresolve.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class BeyondcorpProjectsLocationsAppConnectionsResolvePathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=parent" })
  parent: string;
}


export class BeyondcorpProjectsLocationsAppConnectionsResolveQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=appConnectorId" })
  appConnectorId?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=pageToken" })
  pageToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class BeyondcorpProjectsLocationsAppConnectionsResolveSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class BeyondcorpProjectsLocationsAppConnectionsResolveRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: BeyondcorpProjectsLocationsAppConnectionsResolvePathParams;

  @Metadata()
  queryParams: BeyondcorpProjectsLocationsAppConnectionsResolveQueryParams;

  @Metadata()
  security: BeyondcorpProjectsLocationsAppConnectionsResolveSecurity;
}


export class BeyondcorpProjectsLocationsAppConnectionsResolveResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  googleCloudBeyondcorpAppconnectionsV1alphaResolveAppConnectionsResponse?: shared.GoogleCloudBeyondcorpAppconnectionsV1alphaResolveAppConnectionsResponse;

  @Metadata()
  statusCode: number;
}
